### PR TITLE
Add failsafe check.

### PR DESCRIPTION
**Pull request by qdpham13**
- Add backend failsafe check.
- Add app id and sdk version to request
- add boolean check to response
- [go/realtime-rc-sdk-failsafe](http://go/realtime-rc-sdk-failsafe)

### CODE REVIEW

File: firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigAutoFetch.java [hunk 1]
```diff
@@ -119,7 +119,8 @@ private void handleNotifications(InputStream inputStream) throws IOException {
     BufferedReader reader = new BufferedReader((new InputStreamReader(inputStream, "utf-8")));
     String message;
     while ((message = reader.readLine()) != null) {
-      if (!message.contains("latestTemplateVersionNumber")) {
+      if (!message.contains("latestTemplateVersionNumber")
+          && !message.contains("featureDisabled")) {
```
Comment (rlazo) [inline firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigAutoFetch.java:123]: Without more context I'd assume that the `featureDisabled` would be checked first to short-cut the rest of the cut. Why is it checked last?

**Review by rlazo — COMMENTED** [review]
Would it make sense to add tests for this?

File: firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigAutoFetch.java (hunk 1 above)
Comment (qdpham13) [reply firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigAutoFetch.java:123]: I added the one pager for this change to the description. This flag will always be in the response so this check skips any other messages we may get; it doesn't matter the order of the check.

Comment (qdpham13) [reply firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigAutoFetch.java:123]: Also added tests.

**Pull request closed without merging**